### ISSUE TITLE
Interlock Rectilinear Infill

### ISSUE THREAD

**Issue by espr14** [supermerill/SuperSlicer#571]
More here: https://github.com/prusa3d/PrusaSlicer/issues/4867

**Comment by supermerill** [comment]
biggest issue: how to have a nice flat layer for doing the top infill ?

**Comment by espr14** [comment]
Good point. The last infill layer will behave like 50% infill. Solutions:
1. Do nothing. 50% is far more than e.g. 20% and there's no issue with top layers. You just need at least 2 top layers.
2. Print every other line (odd or even) with 100% flow (like the first infill layer). Direction and odd/even lines will be according to modulo 4 result (direction changes every layer, oddness/evenness changes per two layers).
